profile/src/wallet_kit_common/types/hex_32bytes.rs

```diff
@@ -0,0 +1,254 @@
+use crate::prelude::*;
+use radix_engine_common::crypto::{Hash, IsHash};
+
+/// Serializable 32 bytes which **always** serializes as a **hex** string, this is useful
+/// since in Radix Wallet Kit we almost always want to serialize bytes into hex and this
+/// allows us to skip using
+#[derive(
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    SerializeDisplay,
+    DeserializeFromStr,
+    derive_more::Display,
+    derive_more::Debug,
+    uniffi::Record,
+)]
+#[display("{}", self.to_hex())]
+#[debug("{}", self.to_hex())]
+pub struct Hex32Bytes {
```

Comment:
FIXME: check again if we can use newtype and `uniffi::custom_newtype` with statically size arrays.

Reply:
Tried again and failed, I think UniFFI simply cannot support statically sized arrays - because Swift does not support them!